File: processors/php/commands.php
<?php
/**
 * Regroups all commands for the PHP version of the default game
 */

class GameConsole {
    function print($value) {
        echo "console.log('{$value}');";
    }
}


/**
 * Character of the game
 * Allows you to control its movements
 */
class Pacman {

    /**
     * Moves the character to the left of his position
     *
     * @return void|string
     */
    function moveLeft() {
        fwrite(STDERR, "none\n");
        fflush(STDERR);

        fwrite(STDOUT, "pacman.moveLeft();\n");
        fflush(STDOUT);

        $stdin = fgets(STDIN);

        fwrite(STDOUT, $stdin."\n");
        fflush(STDOUT);

        return $stdin;
    }

    /**
     * Moves the character to the right of his position
     *
     * @return void|string
     */
    function moveRight() {
        fwrite(STDERR, "none\n");
        fflush(STDERR);

        fwrite(STDOUT, "pacman.moveRight();\n");
        fflush(STDOUT);

        $stdin = fgets(STDIN);

        fwrite(STDOUT, $stdin."\n");
        fflush(STDOUT);

        return $stdin;
    }

    /**
     * Moves the character above his position
     *
     * @return void|string
     */
    function moveUp() {
        fwrite(STDERR, "none\n");
        fflush(STDERR);

        fwrite(STDOUT, "pacman.moveUp();\n");
        fflush(STDOUT);

        $stdin = fgets(STDIN);

        fwrite(STDOUT, $stdin."\n");
        fflush(STDOUT);

        return $stdin;
    }

    /**
     * Moves the character below his position
     *
     * @return void|string
     */
    function moveDown() {
        fwrite(STDERR, "none\n");
        fflush(STDERR);

        fwrite(STDOUT, "pacman.moveDown();\n");
        fflush(STDOUT);

        $stdin = fgets(STDIN);

        fwrite(STDOUT, $stdin."\n");
        fflush(STDOUT);

        return $stdin;
    }
}

/**
 * Check if the case on the left of the character is free
 *
 * @return boolean
 */
function isLeftSideFree() {
    fwrite(STDERR, "none\n");
    fflush(STDERR);

    fwrite(STDOUT, "game.isLeftSideFree();\n");
    fflush(STDOUT);

    $stdin = fgets(STDIN);

    fwrite(STDOUT, $stdin."\n");
    fflush(STDOUT);

    return $stdin;
}

/**
 * Check if the case on the right of the character is free
 *
 * @return boolean
 */
function isRightSideFree() {
    fwrite(STDERR, "none\n");
    fflush(STDERR);

    fwrite(STDOUT, "game.isRightSideFree();\n");
    fflush(STDOUT);

    $stdin = fgets(STDIN);
    fwrite(STDOUT, $stdin);

    fwrite(STDOUT, $stdin."\n");
    fflush(STDOUT);

    return $stdin;
}

/**
 * Check if the case on the top of the character is free
 *
 * @return boolean
 */
function isUpSideFree() {
    fwrite(STDERR, "none\n");
    fflush(STDERR);

    fwrite(STDOUT, "game.isUpSideFree();\n");
    fflush(STDOUT);

    $stdin = fgets(STDIN);

    fwrite(STDOUT, $stdin."\n");
    fflush(STDOUT);

    return $stdin;
}

/**
 * Check if the case on the bottom of the character is free
 *
 * @return boolean
 */
function isDownSideFree() {
    fwrite(STDERR, "none\n");
    fflush(STDERR);

    fwrite(STDOUT, "game.isDownSideFree();\n");
    fflush(STDOUT);

    $stdin = fgets(STDIN);

    fwrite(STDOUT, $stdin."\n");
    fflush(STDOUT);

    return $stdin;
}

File: server/languages/ruby/commands.rb
#
# Regroups all commands for the Ruby version of the default game
#

class GameConsole
  def print(value)
    puts "console.log('#{value}');";
  end
end

#
# Character of the game
# Allows you to control its movements
#
class Pacman

  # Moves the character to the left of his position
  def moveLeft
    STDERR.puts "none"
    STDERR.flush

    puts "pacman.moveLeft();"
    STDOUT.flush
    msg = STDIN.gets
    puts msg

    msg
  end

  # Moves the character to the right of his position
  def moveRight
    STDERR.puts "none"
    STDERR.flush

    puts "pacman.moveRight();"
    STDOUT.flush
    msg = STDIN.gets
    puts msg

    msg
  end

  # Moves the character above his position
  def moveUp
    STDERR.puts "none"
    STDERR.flush

    puts "pacman.moveUp();"
    STDOUT.flush
    msg = STDIN.gets
    puts msg

    msg
  end

  # Moves the character below his position
  def moveDown
    STDERR.puts "none"
    STDERR.flush

    puts "pacman.moveDown();"
    STDOUT.flush
    msg = STDIN.gets
    puts msg

    msg
  end

end

# Check if the case on the left of the character is free
#
#   isLeftSideFree
#   # => true|false
#
def isLeftSideFree
  STDERR.puts "none"
  STDERR.flush

  puts 'isLeftSideFree();'
  STDOUT.flush
  msg = STDIN.gets
  puts msg

  msg
end

# Check if the case on the right of the character is free
#
#   isRightSideFree
#   # => true|false
#
def isRightSideFree
  STDERR.puts "none"
  STDERR.flush

  puts 'isRightSideFree();'
  STDOUT.flush
  msg = STDIN.gets
  puts msg

  msg
end

# Check if the case on the top of the character is free
#
#   isUpSideFree
#   # => true|false
#
def isUpSideFree
  STDERR.puts "none"
  STDERR.flush

  puts 'isUpSideFree();'
  STDOUT.flush
  msg = STDIN.gets
  puts msg

  msg
end

# Check if the case on the bottom of the character is free
#
#   isDownSideFree
#   # => true|false
#
def isDownSideFree
  STDERR.puts "none"
  STDERR.flush

  puts 'isDownSideFree();'
  STDOUT.flush
  msg = STDIN.gets
  puts msg

  msg
end
